src/poetry_plugin_export/walker.py

```diff
@@ -68,10 +70,16 @@ def get_project_dependency_packages(
     extra_package_names: set[str] | None = None
 
     if extras is not True:
+        locked_extras = {
+            canonicalize_name(extra): [
+                canonicalize_name(dependency) for dependency in dependencies
+            ]
+            for extra, dependencies in locker.lock_data.get("extras", {}).items()
+        }
         extra_package_names = set(
             get_extra_package_names(
                 repository.packages,
-                locker.lock_data.get("extras", {}),
+                locked_extras,  # type: ignore[arg-type]
```

Comment:
`type: ignore` to avoid `error: Argument 2 to "get_extra_package_names" has incompatible type "Dict[NormalizedName, List[NormalizedName]]"; expected "Mapping[str, List[str]]"  [arg-type]`

Will become unused with updated types in python-poetry/poetry#6541.